packages/@okta/vuepress-site/docs/guides/migrate-to-oie/index.md

```diff
@@ -0,0 +1,155 @@
+    <div id="okta-login-container"></div>
+
+    <!--
+        "OktaUtil" defines a global OktaUtil object
+        that contains methods used to complete the Okta login flow.
+     -->
+    {{{OktaUtil}}}
+
+    <script type="text/javascript">
+        // "config" object contains default widget configuration
+        // with any custom overrides defined in your admin settings.
+        var config = OktaUtil.getSignInWidgetConfig();
+
+        // Render the Okta Sign-In Widget
+        var oktaSignIn = new OktaSignIn(config);
+        oktaSignIn.renderEl({ el: '#okta-login-container' },
+            OktaUtil.completeLogin,
+            function(error) {
+                // Logs errors that occur when configuring the widget.
+                // Remove or replace this with your own custom error handler.
+                console.log(error.message, error);
+            }
+        );
+    </script>
+</body>
+</html>
+```
+
+### Embedded signin widget
+
+Set the option `useInteractionCodeFlow` to `true` on the object passed to the Sign-In Widget constructor. This will enable Okta Identity Engine for the widget. Both the authorization server and the application must have the [interaction code](#enable-interaction-code-grant) grant type enabled.
+
+> **Note:** Your code may break if it is calling the `renderEl` method and expects `sessionToken` or `session.setCookieAndRedirect` on the response object. Instead of `renderEl`, we recommend calling the `showSignInToGetTokens` method. This method receives and returns tokens without any browser redirect.
+
+```javascript
+var signIn = new OktaSignIn(
+  {
+    baseUrl: 'https://{yourOktaDomain}',
+    clientId: '{{clientId of your OIDC app}}'
+    redirectUri: '{{redirectUri configured in OIDC app}}'
+    useInteractionCodeFlow: true
+    // other options...
+  }
+);
+
+signIn.showSignInToGetTokens({
+  // Assumes there is an empty element on the page with an id of 'osw-container'
+  el: '#osw-container'
+}).then(function(tokens) {
+  // Store tokens
+  signIn.authClient.tokenManager.setTokens(tokens);
+  // Remove widget
+  signIn.remove();
+  // Can now render in an authenticated state
+}).catch(function(error) {
+  // Handle error
+})
+```
+
+### Direct authentication
+
+Javascript clients, whether browser-based or running in nodeJS, can use the [okta-idx-js](https://github.com/okta/okta-idx-js) and [okta-auth-js](https://github.com/okta/okta-auth-js) SDKs to authenticate using the interaction code flow.
+
+```javascript
+// Get PKCE params
+const params = await oktaAuth.token.prepareTokenParams();
+const { codeVerifier, codeChallenge, codeChallengeMethod } = params;
+
+// Start IDX
+const idxState = await idx.start({
+  issuer,
+  clientId,
+  redirectUri,
+  // ... other params for IDX
+  // PKCE code challenge is needed for call to /interact
+  codeChallenge,
+  codeChallengeMethod,
+});
+
+// When idx reaches success state, use the interactionCode to obtain tokens
```

Comment:
There's a big step we're skimming over there - that they have to take the responses, build the UI to gather their data (Without losing their state, if they are in the browser), send that using `idxState.proceed()`, and repeat that series until your below code.

We should definitely call that out more, and at least separate the above code from the below code, since we're omitting that work.

Reply:
That was the part I was hoping you could help fill out, a short end to end sample